server/utils/hollaex-tools-lib/tools/security.js

```diff
@@ -841,6 +847,90 @@ const verifyHmacTokenPromise = (apiKey, apiSignature, apiExpires, method, origin
 	}
 };
 
+
+const createSession = async (token, loginId, userId) => {
+
+	const { getUserRole } = require('./user');
+
+	const userRole = await getUserRole({ kit_id: userId });
+
+	const base64Payload = token.split(".")[1];
+	const payloadBuffer = Buffer.from(base64Payload, "base64");
+	const decoded = JSON.parse(payloadBuffer.toString());
+
+	const hashedToken = crypto.createHash('md5').update(token).digest('hex');
+
+	return getModel('session').create({
+		token: hashedToken,
+		role: userRole,
+		login_id: loginId,
+		status: true,
+		last_seen: new Date(),
+		expiry_date: new Date(decoded.exp * 1000)
+	})
+}
+
+const verifySession = async (token) => {
+
+	const session = await findSession(token);
+
+	if (!session || !session.status) {
+		loggerAuth.error(
+			'security/verifySession invalid session',
+			session.status
+		);
+		throw new Error(TOKEN_REVOKED);
+	}
+
+	if(new Date(session.expiry_date).getTime() < new Date().getTime()) {
+		throw new Error(TOKEN_EXPIRED);
+	}
+
+	if(new Date(session.last_seen).getTime() + 1000 * 60 * 5 < new Date().getTime()) {
+		const sessionData = await dbQuery.findOne('session', { where: { token } });
+		const updatedSession =  await sessionData.update(
+			{ last_seen: new Date() }
+		);
+		client.hsetAsync(SESSION_TOKEN_KEY, updatedSession.dataValues.token, JSON.stringify(updatedSession.dataValues));
+	}
+}
+
+const findSession = async (token) => {
+
+	const hashedToken = crypto.createHash('md5').update(token).digest('hex');
+
+	let session = await client.hgetAsync(SESSION_TOKEN_KEY, hashedToken)
+	
+	if (!session) {
+		loggerAuth.verbose(
+			'security/findSession jwt token not found in redis',
+			hashedToken
+		);
+
+		session = await dbQuery.findOne('session', {
+			where: {
+				token: hashedToken
+			}
+		});
+
+		if(session && session.status && new Date(session.expiry_date).getTime() > new Date().getTime())
+			client.hsetAsync(SESSION_TOKEN_KEY, hashedToken, JSON.stringify(session));
```

Comment:
Why dont you create this on redis with expiry. We dont delete these tokens and they dont have any expiry so they just grow constantly this way which is not really good @fetok12